src/coreservices.h

```diff
@@ -35,7 +38,7 @@ class CoreServices : public QObject {
 
   public:
     CoreServices(const CmdlineArgs& args);
-    ~CoreServices() = default;
+    ~CoreServices();
```

Comment:
Why did you move this to the .cpp file?

Reply:
I'm using a `std::unique_ptr` for `ControlIndicatorTimer`. In case we destruct `CoreServices` without calling `shutdown()` first (shouldn't happen, but the compiler does not know this), the compiler needs the full declaration of `ControlIndicatorTimer` to be able to destruct `ControlIndicatorTimer`.

I didn't want to include `controlindicatortimer.h` in `coreservices.h`, so I used a forward declaration. If we put the destructor definition into `coreservices.h` instead of `coreservices.cpp`, the full delcaration of `ControlIndicatorTimer` is not known and will lead to a compiler error about calling `sizeof` with an incomplete type.

Reply:
okay